qiskit/circuit/library/standard_gates/rx.py

```diff
@@ -95,7 +95,16 @@ def inverse(self):
 
         :math:`RX(\lambda)^{\dagger} = RX(-\lambda)`
         """
-        return RXGate(-self.params[0])
+        if isinstance(self.params[0], ParameterExpression):
+            if self.params[0].parameters:
+                param_dict = {}
+                for param in self.params[0].parameters:
+                    param_dict[param] = param.name
+                return RXGate(ParameterExpression(param_dict, -self.params[0].get_sympy_expr()))
+            else:
+                return RXGate(-float(self.params[0].get_sympy_expr()))
+        else:
+            return RXGate(-self.params[0])
```

Comment:
I still don't understand why `RXGate(-self.params[0])` isn't identical here?  The additional code doesn't appear to do anything other than just negate the `ParameterExpression` in a very verbose manner, and it's not at all clear to me why this should occur in `RXGate.inverse`, and not `ParameterExpression.__neg__`.

If the issue if that there may be parameters that were already bound, then it should `try: -float(self.params[0]); except TypeError` (no need to `get_sympy_expr()`), and it really shouldn't be accessing private members of `ParameterExpression`, but really this code should not be in `RXGate` at all - that would be a bug we could consider fixing in `ParameterExpression`.